crates/hotshot/src/block_impl.rs

```diff
@@ -0,0 +1,174 @@
+//! This module provides an implementation of the `HotShot` suite of traits.
```

Comment:
Is this file ported from elsewhere or is it 100% new code?

Reply:
It's ported

Reply:
Yeah it's from `demo.rs` (originally `demos/sdemo.rs`).